awscli/customizations/cloudformation/yamlhelper.py

```diff
@@ -54,27 +64,39 @@ def intrinsics_multi_constructor(loader, tag_prefix, node):
     return {cfntag: value}
 
 
+def _dict_representer(dumper, data):
+    return dumper.represent_dict(data.items())
+
+
 def yaml_dump(dict_to_dump):
     """
     Dumps the dictionary as a YAML document
     :param dict_to_dump:
     :return:
     """
+    FlattenAliasDumper.add_representer(OrderedDict, _dict_representer)
     return yaml.dump(
         dict_to_dump,
         default_flow_style=False,
         Dumper=FlattenAliasDumper,
     )
 
 
+def _dict_constructor(loader, node):
+    # Necessary in order to make yaml merge tags work
```

Comment:
According to [CloudFormation template formats](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/template-formats.html), aliases and hash merges are not supported, but I think it's still useful to be able to parse the syntax.